### PR TITLE
messages/MOSDPing.h: drop unused fields

### PR DESCRIPTION
**Pull request by branch-predictor**
For luminous, let's drop unused peer_as_of_epoch and peer_stat fields
from MOSDPing message.
Followup to #15727, not meant to be backported.

Signed-off-by: Piotr Dałek <piotr.dalek@corp.ovh.com>

qa
- [x] rados
- [ ] upgrade/jewel-x

### CODE REVIEW

File: src/messages/MOSDPing.h [hunk 1]
```diff
@@ -96,11 +100,20 @@ class MOSDPing : public Message {
...
     ::encode(fsid, payload);
     ::encode(map_epoch, payload);
-    ::encode(peer_as_of_epoch, payload);
-    ::encode(op, payload);
-    ::encode(peer_stat, payload);
+    
+    // with luminous, we drop peer_as_of_epoch and peer_stat
+    if (!HAVE_FEATURE(features, SERVER_LUMINOUS)) { 
+      epoch_t dummy_epoch = {};
+      osd_peer_stat_t dummy_stat = {};
```
Comment (gregsfortytwo) [inline src/messages/MOSDPing.h]: Is there a reason you're doing the funny initializer?

Yeah, "funny initializer" is how much of some of the newer stuff I really grok. :(

**Comment by branch-predictor** [comment]
On 17-06-22 10:17 PM, Gregory Farnum wrote:
> 
> +      epoch_t dummy_epoch = {};
> +      osd_peer_stat_t dummy_stat = {};
> 
> Is there a reason you're doing the funny initializer?
> 
> Yeah, "funny initializer" is how much of some of the newer stuff I really 
> grok. :(

It's a zero-initializer 
(http://en.cppreference.com/w/cpp/language/zero_initialization), though I 
see now it wasn't really necessary.

-- 
Piotr Dałek

**Comment by liewegas** [comment]
looks ok in rados, http://pulpito.ceph.com/sage-2017-07-03_15:41:59-rados-wip-sage-testing-distro-basic-smithi/
needs upgrade suite since rados/upgrade is broken

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-07-12_02:32:14-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**